### PR TITLE
DS-4218 - Support for second "initialize-entities"

### PR DESCRIPTION
**Pull request by benbosman**
Executing InitializeEntities multiple times to update the schema used to work, but was broken in https://github.com/DSpace/DSpace/pull/2383 by changing the concept of creating relationship types.

The `InitializeEntities` implementation has been updated to solve this problem and a unique constraint to the RelationshipType table has been created based on types and labels

This solves https://jira.duraspace.org/browse/DS-4218

### CODE REVIEW

**Comment by AlexanderS** [comment]
I just opened a pr at atmire/DSpace#15 to fix a minor problem with the indentation.

**Comment by benbosman** [comment]
Thanks @AlexanderS, I've merged it

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        getClient().perform(get("/api/core/relationshiptypes"))
+
+                   //We expect a 200 OK status
+                   .andExpect(status().isOk())
+                   //The type has to be 'discover'
+                   .andExpect(jsonPath("$.page.totalElements", is(10)))
+                   //There needs to be a self link to this endpoint
+                   .andExpect(jsonPath("$._links.self.href", containsString("api/core/relationshiptypes")))
+                   //We have 4 facets in the default configuration, they need to all be present in the embedded section
```
Comment (tdonohue) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java]: incorrect comment

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        alteredRelationshipType.setLeftMinCardinality(10);
+        getClient().perform(get("/api/core/relationshiptypes"))
+
+                   //We expect a 200 OK status
+                   .andExpect(status().isOk())
+                   //The type has to be 'discover'
+                   .andExpect(jsonPath("$.page.totalElements", is(10)))
+                   //There needs to be a self link to this endpoint
+                   .andExpect(jsonPath("$._links.self.href", containsString("api/core/relationshiptypes")))
+                   //We have 4 facets in the default configuration, they need to all be present in the embedded section
```
Comment (tdonohue) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java]: This comment is also incorrect.  Could we better describe this test in general. It's not straightforward what we are looking for... are we simply verifying running this command twice results in a no-op?  Are we looking for specific changes?  This test needs some better comments in general.

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            File.separator + "config" + File.separator + "entities" + File.separator + "relationship-types-update.xml";
+        runDSpaceScript("initialize-entities", "-f", pathToFile);
+
+        RelationshipType alteredRelationshipType = relationshipTypes.get(0);
+        alteredRelationshipType.setLeftMinCardinality(10);
+        getClient().perform(get("/api/core/relationshiptypes"))
+
+                   //We expect a 200 OK status
+                   .andExpect(status().isOk())
+                   //The type has to be 'discover'
```
Comment (tdonohue) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java]: incorrect comment

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+
+
+        List<RelationshipType> relationshipTypes = relationshipTypeService.findAll(context);
+
+        getClient().perform(get("/api/core/relationshiptypes"))
+
+                   //We expect a 200 OK status
+                   .andExpect(status().isOk())
+                   //The type has to be 'discover'
```
Comment (tdonohue) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java]: incorrect comment

**Review by tdonohue — COMMENTED** [review]
@benbosman : the code here looks fine to me, but the new test has some incorrect comments and needs a better description. I'm not sure exactly what it is supposed to be testing.

**Review by paulo-graca — APPROVED** [review]
I've performed a test with this PR and it worked. I can do several successful attempts to import entities structure.

**Review by tdonohue — APPROVED** [review]
Thanks @benbosman and @Raf-atmire .  The updates look great to me now.  Merging into `configurable_entities` branch.

**Pull request merged**